packages/graphql/lib/src/links/websocket_link/websocket_client.dart

```diff
@@ -173,6 +173,7 @@ class SocketClient {
       HashMap();
 
   bool _connectionWasLost = false;
+  bool _wasDisposed = false;
```

Comment:
Why don't add a new enum state?

we could improve also the state pattern, to make more smart changes of state around, but I put here the question that I had during the reading of your PR.

Reply:
You mean another enum state for `_connectionStateController`? one of the potential challenges with that is that the BehaviorSubject won't allow changing the value of it (which has to happen by adding the value to the stream) if the subject (stream controller) has already been closed. since the `_connectionStateController` is only explicitly closed in one place right now, it could ensure that it's set to this new enum value prior to closure, and that nothing else would ever change the state away from that before changing the state. That would result in a lot of other guard conditions where the state is modified throughout the rest of the file, so it seemed more complex. I'm happy to try that approach though if you prefer.

Reply:
No, I like your approach, I was thinking that maybe it is possible to make the state pattern more "pattern" and not with an enum, but this is out of context of this PR.

Thanks